Triton_Inference_Server_Python_API/examples/kafka-io/README.md

```diff
@@ -0,0 +1,136 @@
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Triton Inference Server Kafka I/O Deployment
+
+Using the Triton Inference Server In-Process Python API you can
+integrate triton server based models into any Python framework
+to consume the messages from a Kafka topic and produce the inference results
+back to the kafka topic of choice.
+
+This directory contains an example Triton Inference Server
+deployment based on Kafka I/O that uses threads for each of server, consumer and producer.
+
+| [Installation](#installation) | [Run Deployment](#starting-the-pipeline) | [Send Requests](#send-requests-to-deployment) |
+
+
+## Installation
+
+In this Kafka I/O pipeline we deploy a pre-processing stage of tokenization based on `transformers` tokenization module and can be extended to any type of models as needed.
+
+### Pre-requisite
+1. [Docker](https://docs.docker.com/engine/install/)
+
+### Starting docker container
+Once you have the docker service up and running, launch a container by executing the following command:
+
+```bash
+docker run --rm -it --gpus all -v <path>/<to>/tutorials/Triton_Inference_Server_Python_API/examples/kafka-io/:/opt/tritonserver/kafka-io -w /opt/tritonserver/kafka-io  --entrypoint bash nvcr.io/nvidia/tritonserver:24.06-py3
+```
+
+### Clone Repository
+
+```bash
+git clone https://github.com/triton-inference-server/tutorials.git
+cd tutorials/Triton_Inference_Server_Python_API/examples/kafka-io
+```
+
+*Note: Skip this step if you have mounted the git repository from local directory to the docker container*
+
+
+### Install dependencies
+
+Please note that installation times may vary depending on
+your hardware configuration and network connection.
+
+
+```bash
+pip install -r requirements.txt
+```
+
+If triton server is not already installed, install the dependency by using the following command.
+
+```bash
+pip install /opt/tritonserver/python/tritonserver-2.44.0-py3-none-any.whl
+```
+
+Next run the provided `start-kafka.sh` script that will perform the following actions:
+1. Download kafka and it's dependencies
+2. Start Kafka service by starting Zookeeper and Kafka brokers
+3. Create 2 new topics with names `inference-input` as input queue and `inference-output` to store the inference results
+
+```bash
+chmod +x start-kafka.sh
+./start-kafka.sh
+```
+
+## Starting the pipeline
+
+### Start the inference pipeline
+
+Run the provided `start-server.sh` script that will perform the following actions:
+1. Export Kafka Producer and Consumer configs, topic names for input and output topics, model name and repositories.
+2. Start the server.
+
+```bash
+chmod +x start-server.sh
+./start-server.sh
+```
+
```

Comment:
add:

When your console outputs something similar to:
```
2024-07-18 21:55:38,254 INFO api.py:609 -- Deployed app 'default' successfully.
```
It means that the server has started successfully. You can press `Ctrl+C` and proceed to the next steps.

Reply:
Updated